compiler/rustc_mir_transform/src/check_enums.rs

```diff
@@ -0,0 +1,583 @@
+use rustc_abi::{
+    AddressSpace, Integer, Primitive, Scalar, Size, TagEncoding, Variants, WrappingRange,
+};
+use rustc_hir::LangItem;
+use rustc_index::IndexVec;
+use rustc_middle::bug;
+use rustc_middle::mir::visit::Visitor;
+use rustc_middle::mir::*;
+use rustc_middle::ty::{self, Ty, TyCtxt, TypingEnv};
+use rustc_session::Session;
+use tracing::debug;
+
+/// This pass inserts checks at places where enums are constructed and checks
```

Comment:
Which places? (Also you should probably say "locations" here, it's more similar to the types)

If you don't want to be too specific about where checks are inserted maybe talk around why the pass currently inserts the checks it does. "We insert checks where they are most likely to find UB, because checking everywhere like Miri would generate too much MIR". Or something like that.

Reply:
Makes sense! Thanks!